src/components/Goods/Goods.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { GoodsList } from '../GoodsList/GoodsList';
+import { LoadButtons } from '../LoadButtons/LoadButtons';
+import { Good } from '../../interfaces/good';
+
+interface State {
+  goods: Good[];
+  isLoading: boolean;
+  hasError: boolean;
+}
+
+export class Goods extends React.Component<{}, State> {
+  state = {
+    goods: [],
+    isLoading: false,
+    hasError: false,
+  };
+
+  setLoading = (status: boolean) => {
+    this.setState({
+      isLoading: status,
+    });
+  };
+
+  setError = (status: boolean) => {
+    this.setState({
+      hasError: status,
+    });
+  };
```

Comment:
```suggestion
  setLoading = (status) => {
    this.setState({
      isLoading: status,
    });
  };

  setError = (status) => {
    this.setState({
      hasError: status,
    });
  };
```
в этом случаи ты не можешь записать в эти поля state ничего кроме boolean

Reply:
А typescript так не считает, требует определения типа

Reply:
а тогда ладно)